iep-011/README.adoc

```diff
@@ -0,0 +1,117 @@
+
+=== Endpoint
+The service will listen on two different endpoints:
+
+* 'pkg.jenkins.io' which accept HTTP/HTTPS requests
+* 'download.jenkins.io' which always redirect HTTP to HTTPS
+
+'download.jenkins.io' will be the new default endpoint and for people who explicitly need a HTTP endpoint, pkg.jenkins.io will be provided as a fallback.
+
+=== Service
+The service will be a web proxy deployed on Kubernetes and it will provide packages for all supported distributions.
+
+.Kubernetes Service Schema
+----
+  +----------------------------+
+  | Ingress:                   |
+  |  +----------------------+  |        +--------------------+        +------------------------+
+  |  | /debian              +---------->| Service: Debian    +------->| Deployment: Debian     |
+  |  +----------------------+  |        +--------------------+        +------------------------+
+  |  +----------------------+  |        +--------------------+        +------------------------+
+  |  | /redhat              +---------->| Service: RedHat    +------->| Deployment: RedHat     |
+  |  +----------------------+  |        +--------------------+        +------------------------+
+  |  +----------------------+  |        +--------------------+        +------------------------+
+  |  | /opensuse            +---------->| Service: OpenSuse  +------->| Deployment: Opensuse   |
+  |  +----------------------+  |        +--------------------+        +------------------------+
+  +----------------------------+
+----
+
+IMPORTANT: As we don't want to rely on local mirrors anymore, we must take into account the user experience when downloading packages from pkg.jenkins.io.
+
+== Motivation
+The motivations are multiple:
+
+1. Do not rely on third party link:http://mirrors.jenkins.io/status.html[mirrors].
+2. Use more efficiently our cloud environment.
+3. Have a better control on the packages that we distribute.
+
+== Rationale
+=== Mirrors
+One of the solution would be to deploy http://mirrorbrain.org/[mirrorbrain] and keep relying on third party link:http://mirrors.jenkins.io/status.html[mirrors].
+This wouldn't change too much the current design, excepted that we can use cheaper infrastructure and rely more on the community.
+The downside of this approach are:
+
+* We don't control third party mirrors which means that when issues appear on them, we can't fix them. We can only remove those mirrors from the link:http://mirrors.jenkins.io/status.html[list] until issues are fixed.
+* We are still vulnerable to link:https://www2.cs.arizona.edu/stork/packagemanagersecurity/faq.html[MiTM] attack
+
+=== CDN
+We could use a link:https://azure.microsoft.com/en-us/services/cdn/[CDN] to distribute packages, the only downside is the cost of such services. 
+We still have to evaluate how expensive it would be.
+
+=== One service by region
+We could also duplicate the service in different locations.
```

Comment:
I'm not sure I understand why providing distribution in another region would require a different Kubernetes cluster as opposed to just a change in Azure storage settings?

Perhaps I'm missing something here.

Reply:
This was just an idea that quickly crossed my mind so I was thinking to use an azure traffic manager service https://docs.microsoft.com/en-us/azure/traffic-manager/traffic-manager-overview in front of several <your_region>.pkg.jenkins.io. If we use a web proxy on the kube cluster, this means that we need several kubernetes clusters (one per region).